### ISSUE TITLE
Button nodes should have both valign and halign properties for icons and text

### ISSUE THREAD

**Issue by Shadowblitz16** [godotengine/godot-proposals#375]
**Describe the project you are working on:**
Gameboy Zelda Editor

**Describe the problem or limitation you are having in your project:**
Can't set the halign or valign of icons specifically in buttons

**Describe how this feature / enhancement will help you overcome this problem or limitation:**
adding both vertical and horizontal alignment for text and icons allows the buttons to be more customizable without hackish workarounds 

**Show a mock up screenshots/video or a flow diagram explaining how your proposal will work:**
It doesn't need it

**Describe implementation detail for your proposal (in code), if possible:**
basically it would be the align property of the button with a combobox but it would have 4 of them instead of just 1.
2 for icon alignment (x and y), 2 for text alignment (x and y)

**If this enhancement will not be used often, can it be worked around with a few lines of script?:**
this feature would be used often and idk if it can be done with scripts.
its kinda one of those things that should already be added since every other gui framework has it.

**Is there a reason why this should be core and not an add-on in the asset library?:**
because alignment already is, its just not very useful right now.

**Comment by KoBeWi** [comment]
> Show a mock up screenshots/video or a flow diagram explaining how your proposal will work:
It doesn't need it

It does. You don't even mention specific nodes.

**Comment by Shadowblitz16** [comment]
updated it to mention buttons

**Comment by Shadowblitz16** [comment]
thankyou
